Require NsatzTactic: nsatz support for Z and Q (#12861)
coq/coq: doc/sphinx/addendum/nsatz.rst

   You can load the ``Nsatz`` module with the command ``Require Import Nsatz``.

>    Alternatively, if you prefer to not transitively depend on the
>    files declaring the axioms used to define the real numbers, you can
>    ``Require Import NsatzTactic`` instead; this will still allow
>    :tacn:`nsatz` to solve goals defined about :math:`\mathbb{Z}`,
>    :math:`\mathbb{Q}`, and any user-registered rings.

Reviewer: ```suggestion
   Alternatively, if you prefer not to transitively depend on the
   files declaring the axioms used to define the real numbers, you can
   ``Require Import NsatzTactic`` instead; this will still allow
   :tacn:`nsatz` to solve goals defined about :math:`\mathbb{Z}`,
   :math:`\mathbb{Q}` and any user-registered rings.
```

Is there a simpler phrase than "declaring the axioms used to define"?

Author: Not that I'm aware of

     then `lvar` is replaced by all the variables which are not in
     `parameters`.

> See the file `test-suite/success/Nsatz.v` for many examples, especially in geometry.

Reviewer: Not so convenient for users who haven't downloaded the source tree.  Maybe put in a hyperlink to the file in GitHub?

Author: cc @coq/doc-maintainers , how should I hyperlink to test-suite files?  (and is there a way to version it, or should I just link to current master?)

Reviewer: The markup syntax is like this, in which `Ocaml Gc` is the displayed text (e.g. `Nsatz.v` in your case):

```
`OCaml Gc <http://caml.inria.fr/pub/docs/manual-ocaml/libref/Gc.html#VALcompact>`_
```

Even if GitHub allows linking to a specific version of the file (I'm guessing it does), we don't yet have a mechanism/procedure to update such links in future releases of Coq.  Maybe we should create that.  Not sure how many places in the doc we would use that feature.  @Zimmi48, your thoughts?

The words "the file" seem unnecessary, BTW.

Author: @jfehrle So like this?
```suggestion
See `Nsatz.v in the test-suite <https://github.com/coq/coq/blob/master/test-suite/success/Nsatz.v>`_ for many examples, especially in geometry.
```
Is there a way to get `Nsatz.v` to still be teletype font even as part of a link?

Reviewer: > Is there a way to get `Nsatz.v` to still be teletype font even as part of a link?

Not that I'm aware of.  My impression is that a lot of the markup operations are not orthogonal and can't be combined.

Maybe move "in the test-suite" so it's after and not part of the link.

Reviewer 2: I think linking to master is fine. We don't have that many links so far.

Author: ```suggestion
See the test-suite file `Nsatz.v <https://github.com/coq/coq/blob/master/test-suite/success/Nsatz.v>`_ for many examples, especially in geometry.
```
